### PR TITLE
theming: fix decoration color theming

### LINKED ISSUE #6839: [theming] minimap is incorrectly decorated for scm updates (additions, modifications)
**Description**

The minimap is incorrectly decorated for modified, and added lines (always red).
The minimap should be correctly decorated similarly to the `diff-editor`.

_Single Editor:_

<div align='center'>

editor

</div>

_Diff Editor:_

<div align='center'>

good

</div>

**Steps to Reproduce:**
1. open a workspace under source control
2. ensure that minimap is enabled
3. perform additions, deletions, modifications
4. notice that the minimap is incorrectly decorated

### PR DESCRIPTION
**Pull request by vince-fugnitto**
#### What it does

Fixes #6839

The following commit addresses the following problems:
- [x] fixes the `DecorationOptions` [color](https://github.com/eclipse-theia/theia/pull/7330/files#diff-3847bbcc53225a0d03be91200806e7beR91) to also accept the `id` of color variables.
- [x] fixes the incorrect application of decorations for `scm`
(includes **additions**, **deletions**, and **modifications**).
- [x] fixes the color variables for `editorOverviewRuler`, aligning the values with those of vscode.
- [x] aligns the fix to other areas of the framework, notably `search-in-workspace`.

#### How to test

1. start the application with a workspace under version control (git).
2. verify that **additions**, **modifications**, **deletions** are properly themed (similarly to vscode).
3. change themes, and verify that colors are properly applied (ex: custom [git-neon](https://github.com/vince-fugnitto/git-neon/releases/download/1.0.0/git-neon-0.0.1.vsix)).
4. perform a search-in-workspace search, verify that results are properly themed.

<div align='center'>

Screen Shot 2020-04-24 at 10 46 45 AM

</div>

#### Review checklist

- [x] as an author, I have thoroughly tested my changes and carefully followed [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#requesting-a-review)

#### Reminder for reviewers

- as a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)

Signed-off-by: Vincent Fugnitto <vincent.fugnitto@ericsson.com>

### CODE REVIEW

File: packages/scm/src/browser/scm-contribution.ts [hunk 1]
```diff
@@ -233,17 +238,23 @@ export class ScmContribution extends AbstractViewContribution<ScmWidget> impleme
             },
             {
                 id: 'editorOverviewRuler.modifiedForeground', defaults: {
-                    dark: overviewRulerDefault, light: overviewRulerDefault, hc: overviewRulerDefault
+                    dark: Color.transparent(SCM_COLORS.editorGutterModifiedBackground, 0.6),
```
Comment (vince-fugnitto) [inline packages/scm/src/browser/scm-contribution.ts]: https://github.com/microsoft/vscode/blob/a906625efbc2d0b19ff27eb870e478a047e431e3/src/vs/workbench/contrib/scm/browser/dirtydiffDecorator.ts#L845-L847

**Comment by vince-fugnitto** [comment]
I created a custom theme [git-neon](https://github.com/vince-fugnitto/git-neon/releases/download/1.0.0/git-neon-0.0.1.vsix) to verify if the colors were successfully applied.

Screen Shot 2020-03-12 at 6 34 10 PM

File: packages/scm/src/browser/scm-contribution.ts [hunk 2]
```diff
@@ -53,6 +53,12 @@ export namespace SCM_COMMANDS {
     };
 }
 
+export namespace SCM_COLORS {
```
Comment (akosyakov) [inline packages/scm/src/browser/scm-contribution.ts]: `ScmColors` it is not a constant

File: packages/scm/src/browser/dirty-diff/dirty-diff-decorator.ts [hunk 3]
```diff
@@ -92,4 +57,64 @@ export class DirtyDiffDecorator extends EditorDecorator {
...
+    /**
+     * Get the modified line decoration.
+     *
+     * @returns the editor decoration options for modifications.
+     */
+    protected getModifiedLineDecorationOptions(): EditorDecorationOptions {
+        return <EditorDecorationOptions>{
+            linesDecorationsClassName: 'dirty-diff-glyph dirty-diff-modified-line',
+            overviewRuler: {
+                color: this.colorRegistry.getCurrentColor('editorOverviewRuler.modifiedForeground'),
```
Comment (akosyakov) [inline packages/scm/src/browser/dirty-diff/dirty-diff-decorator.ts]: It does not look good, it has to be variable name. We register all variables for monaco with proper colors already. If we do it like that we hard code a concrete value at some point of time. It is bogus, e.g. try to change theme after colors are applied.

Comment (vince-fugnitto) [reply packages/scm/src/browser/dirty-diff/dirty-diff-decorator.ts]: @akosyakov which part needs to be a variable name, the color?
Unfortunately, if we don't try and get the current color value from the registry, no color is properly applied. Perhaps the change needs to be higher to apply colors properly from `ids`.

Comment (akosyakov) [reply packages/scm/src/browser/dirty-diff/dirty-diff-decorator.ts]: I thought `color: 'mycolor.variable'` should work, Monaco will internally use our color registry to resolve it, we rely on it in other places as well, i.e. in the plugin system

Maybe our Monaco version is old for it?

Comment (akosyakov) [reply packages/scm/src/browser/dirty-diff/dirty-diff-decorator.ts]: No Monaco has it already for 3 years: https://github.com/microsoft/vscode/commit/a42ad12d9c0efd4376884c41a2fb71a077a42b13#diff-285a87c3190bca33a7bdd2b1ce3e9c8aR20-R23

Something wrong with our color registration.

Comment (vince-fugnitto) [reply packages/scm/src/browser/dirty-diff/dirty-diff-decorator.ts]: > No Monaco has it already for 3 years: [microsoft/vscode@a42ad12#diff-285a87c3190bca33a7bdd2b1ce3e9c8aR20-R23](https://github.com/microsoft/vscode/commit/a42ad12d9c0efd4376884c41a2fb71a077a42b13#diff-285a87c3190bca33a7bdd2b1ce3e9c8aR20-R23)
> 
> Something wrong with our color registration.

I did a bit of investigation today as to why the registration for these items (ruler and minimap) and could not identify the root problem.

Comment (vince-fugnitto) [reply packages/scm/src/browser/dirty-diff/dirty-diff-decorator.ts:70]: @akosyakov any idea as to why the monaco color registration (particularly for 'overviewRuler' and 'minimap') is not working correctly? I haven't been able to identify the root cause and the bug is quite annoying when trying to see git changes but everything is red.

Comment (akosyakov) [reply packages/scm/src/browser/dirty-diff/dirty-diff-decorator.ts:70]: @vince-fugnitto it should be: 
```ts
const RemovedLineDecoration = <EditorDecorationOptions>{
    linesDecorationsClassName: 'dirty-diff-glyph dirty-diff-removed-line',
    overviewRuler: {
        color: {
            id: 'editorOverviewRuler.deletedForeground'
        },
        position: OverviewRulerLane.Left,
    },
    minimap: {
        color: {
            id: 'minimapGutter.deletedBackground'
        },
        position: MinimapPosition.Gutter
    },
    isWholeLine: false
};
```

out typings in the editor extension are outdated

Comment (vince-fugnitto) [reply packages/scm/src/browser/dirty-diff/dirty-diff-decorator.ts]: Thank you @akosyakov!

**Pull request merged**